src/io/flutter/logging/FlutterConsoleLogManager.java

```diff
@@ -144,4 +172,54 @@ private void printStackTraceToConsole(
   private String stringValueFromStringRef(InstanceRef ref) {
     return ref.getValueAsStringIsTruncated() ? ref.getValueAsString() + "..." : ref.getValueAsString();
   }
+
+  private String stringValueFromStringRef(Instance instance) {
+    return instance.getValueAsStringIsTruncated() ? instance.getValueAsString() + "..." : instance.getValueAsString();
+  }
+
+  private String getFullStringValue(@NotNull VmService service, String isolateId, @Nullable InstanceRef ref) {
+    if (ref == null) return null;
+
+    if (!ref.getValueAsStringIsTruncated()) {
+      return ref.getValueAsString();
+    }
+
+    final CountDownLatch latch = new CountDownLatch(1);
+    final String[] result = new String[1];
```

Comment:
Why a length one array instead of a simple string?

Reply:
I think because `result` gets assigned in the lambda.  (There is probably a more modern idiom for this though...)

Reply:
Yup, it’s a Java idiom. Hopefully not too out of date…